graph/pull_v2.go

```diff
@@ -223,6 +223,9 @@ func (p *v2Puller) pullV2Tag(tag, taggedName string) (verified bool, err error)
 	go func() {
 		if _, err := io.Copy(out, pipeReader); err != nil {
 			logrus.Errorf("error copying from layer download progress reader: %s", err)
+			if _, err := io.Copy(ioutil.Discard, pipeReader); err != nil {
```

Comment:
Ok, for `v1.8.2` we don't want to change the download logic, only make it compatible with `v1.7.1` and fix the blocking. I'll change the `pipeReader` to close.